model-engine/model_engine_server/common/dtos/llms.py

```diff
@@ -87,14 +87,62 @@ class GetLLMModelEndpointV1Response(BaseModel):
     inference_framework_image_tag: Optional[str] = None
     num_shards: Optional[int] = None
     quantize: Optional[Quantization] = None
+    checkpoint_path: Optional[str] = None
     spec: Optional[GetModelEndpointV1Response] = None
 
 
 class ListLLMModelEndpointsV1Response(BaseModel):
     model_endpoints: List[GetLLMModelEndpointV1Response]
 
 
-# Delete and update use the default Launch endpoint APIs.
+class UpdateLLMModelEndpointV1Request(BaseModel):
+    # LLM specific fields
+    model_name: Optional[str]
+    source: Optional[LLMSource]
+    inference_framework: Optional[LLMInferenceFramework]
+    inference_framework_image_tag: Optional[str]
+    num_shards: Optional[int]
+    """
+    Number of shards to distribute the model onto GPUs. Only affects behavior for text-generation-inference models
+    """
+
+    quantize: Optional[Quantization]
+    """
+    Whether to quantize the model. Only affect behavior for text-generation-inference models
+    """
+
+    checkpoint_path: Optional[str]
+    """
+    Path to the checkpoint to load the model from. Only affects behavior for text-generation-inference models
+    """
+
+    # General endpoint fields
+    metadata: Optional[Dict[str, Any]]
+    post_inference_hooks: Optional[List[str]]
+    endpoint_type: Optional[ModelEndpointType]
```

Comment:
Hmm ok below, you do throw a runtime error to prevent this. Makes me wonder why in Launch we even allowed `endpoint_type` in the request payload for updates 🤔